soco/core.py

```diff
@@ -1578,6 +1580,246 @@ def _build_album_art_full_uri(self, url):
+        count = self.music_library.browse(ml_item=sonos_playlist).total_matches
+        tracks = ','.join([str(x) for x in range(count)])
+        if tracks:
+            return self.reorder_sonos_playlist(sonos_playlist, tracks=tracks,
+                                               new_pos='', update_id=update_id)
+        else:
+            return {'change': 0, 'update_id': update_id, 'length': count}
+
+    @only_on_master
+    def move_in_sonos_playlist(self, sonos_playlist, track, new_pos,
+                               update_id=0):
+        """Move a track to a new position within a Sonos Playlist.
+        This is a convenience method for :py:meth:`reorder_sonos_playlist`.
+
+        Example::
+
+            device.move_in_sonos_playlist(sonos_playlist, track=0, new_pos=1)
+
+        Args:
+            sonos_playlist
+                (:py:class:`~.soco.data_structures.DidlPlaylistContainer`):
+                Sonos playlist object or the item_id (str) of the Sonos
+                playlist.
+            track (int): **0**-based position of the track to move. The first
+                track is track 0, just like indexing into a Python list.
+            new_pos (int): **0**-based location to move the track.
+            update_id (int): Optional update counter for the object. If left
+                at the default of 0, it will be looked up.
+
+        Returns:
+            dict: See :py:meth:`reorder_sonos_playlist`
+
+        Raises:
+            SoCoUPnPException: See :py:meth:`reorder_sonos_playlist`
+        """
+        return self.reorder_sonos_playlist(sonos_playlist, int(track),
+                                           int(new_pos), update_id)
+
+    @only_on_master
+    def remove_from_sonos_playlist(self, sonos_playlist, track, update_id=0):
+        """Remove a track from a Sonos Playlist.
+        This is a convenience method for :py:meth:`reorder_sonos_playlist`.
+
+        Example::
+
+            device.remove_from_sonos_playlist('SQ:0', track=0)
+
+        Args:
+            sonos_playlist
+                (:py:class:`~.soco.data_structures.DidlPlaylistContainer`):
+                Sonos playlist object or the item_id (str) of the Sonos
+                playlist.
+            track (int): *0**-based position of the track to move. The first
+                track is track 0, just like indexing into a Python list.
+            update_id (int): Optional update counter for the object. If left
+                at the default of 0, it will be looked up.
+
+        Returns:
+            dict: See :py:meth:`reorder_sonos_playlist`
+
+        Raises:
+            SoCoUPnPException: See :py:meth:`reorder_sonos_playlist`
+        """
+        return self.reorder_sonos_playlist(sonos_playlist, int(track), None,
+                                           update_id)
+
+    @only_on_master
+    def get_sonos_playlist_by_attr(self, attr_name, match):
+        """ Return the first Sonos Playlist DidlPlaylistContainer that
```

Comment:
done

On Mon, Sep 14, 2015 at 7:43 AM, Kenneth Nielsen <notifications@github.com>
wrote:

> In soco/core.py
> <https://github.com/SoCo/SoCo/pull/352#discussion_r39388508>:
>
> > +                track is track 0, just like indexing into a Python list.
> > +            update_id (int): Optional update counter for the object. If left
> > +                at the default of 0, it will be looked up.
> > +
> > +        Returns:
> > +            dict: See :py:meth:`reorder_sonos_playlist`
> > +
> > +        Raises:
> > +            SoCoUPnPException: See :py:meth:`reorder_sonos_playlist`
> > +        """
> > +        return self.reorder_sonos_playlist(sonos_playlist, int(track), None,
> > +                                           update_id)
> > +
> > +    @only_on_master
> > +    def get_sonos_playlist_by_attr(self, attr_name, match):
> > +        """ Return the first Sonos Playlist DidlPlaylistContainer that
>
> No space before "Return"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SoCo/SoCo/pull/352/files#r39388508>.
>



-- 
Best,

Jeff Hinrichs
dundeemt@gmail.com
402.218.1473